--- a/paper-server/build.gradle.kts
+++ b/paper-server/build.gradle.kts
@@ -22,9 +_,23 @@
     minecraftVersion = providers.gradleProperty("mcVersion")
     gitFilePatches = false
 
+    // Pufferfish start - Rebrand
+    val pufferfish = forks.register("pufferfish") {
+        upstream.patchDir("paperServer") {
+            upstreamPath = "paper-server"
+            excludes = setOf("src/minecraft", "patches", "build.gradle.kts")
+            patchesDir = rootDirectory.dir("pufferfish-server/paper-patches")
+            outputDir = rootDirectory.dir("paper-server")
+        }
+    }
+    activeFork = pufferfish
+    // Pufferfish end - Rebrand
+
+    /* Pufferfish - Unneeded
     updatingMinecraft {
         // oldPaperCommit = "d4fe85375af18bfa88f44d7c1e6a61904ae550cc"
     }
+     */
 }
 
 tasks.generateDevelopmentBundle {
@@ -86,7 +_,21 @@
     }
 }
 
-val log4jPlugins = sourceSets.create("log4jPlugins")
+// Pufferfish start - Rebrand
+sourceSets {
+    main {
+        java { srcDir("../paper-server/src/main/java") }
+        resources { srcDir("../paper-server/src/main/resources") }
+    }
+    test {
+        java { srcDir("../paper-server/src/test/java") }
+        resources { srcDir("../paper-server/src/test/resources") }
+    }
+}
+val log4jPlugins = sourceSets.create("log4jPlugins") {
+    java { srcDir("../paper-server/src/log4jPlugins/java") }
+}
+// Pufferfish end - Rebrand
 configurations.named(log4jPlugins.compileClasspathConfigurationName) {
     extendsFrom(configurations.compileClasspath.get())
 }
@@ -109,7 +_,7 @@
 }
 
 dependencies {
-    implementation(project(":paper-api"))
+    implementation(project(":pufferfish-api")) // Pufferfish
     implementation("ca.spottedleaf:leafpile:1.0.0")
     implementation("org.jline:jline-terminal-ffm:3.27.1") // use ffm on java 22+
     implementation("org.jline:jline-terminal-jni:3.27.1") // fall back to jni on java 21
@@ -144,6 +_,10 @@
         isTransitive = false // includes junit
     }
 
+    implementation("net.serlith:Flare:4.1.5") // Pufferfish - Flare
+    implementation("net.serlith.jshepherd:core:4.1.0") // Pufferfish - Config Library
+    implementation("net.serlith.jshepherd:yaml:4.1.0") // Pufferfish - Config Library
+
     testImplementation("io.github.classgraph:classgraph:4.8.184") // For mob goal test
     testRuntimeOnly("org.junit.platform:junit-platform-launcher:6.0.3")
     testImplementation("org.junit.jupiter:junit-jupiter:6.0.3")
@@ -155,8 +_,8 @@
     testImplementation("org.junit-pioneer:junit-pioneer:2.3.0") // CartesianTest
 
     // Spark
-    implementation("me.lucko:spark-api:0.1-20240720.200737-2")
-    implementation("me.lucko:spark-paper:1.10.172")
+    implementation("me.biquaternions.spark:spark-api:1.5-pufferfish")
+    implementation("me.biquaternions.spark:spark-paper:1.5-pufferfish")
 }
 
 tasks.jar {
@@ -171,14 +_,14 @@
         val gitBranch = git.exec(providers, "rev-parse", "--abbrev-ref", "HEAD").get().trim()
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
-            "Implementation-Title" to "Paper",
+            "Implementation-Title" to "Puffernot", // Pufferfish
             "Implementation-Version" to implementationVersion,
             "Implementation-Vendor" to date,
-            "Specification-Title" to "Paper",
+            "Specification-Title" to "Puffernot", // Pufferfish
             "Specification-Version" to project.version,
-            "Specification-Vendor" to "Paper Team",
-            "Brand-Id" to "papermc:paper",
-            "Brand-Name" to "Paper",
+            "Specification-Vendor" to "Pufferfish Host & Serlith Network", // Pufferfish
+            "Brand-Id" to "pufferfish:pufferfish", // Pufferfish
+            "Brand-Name" to "Pufferfish", // Pufferfish
             "Build-Number" to (build ?: ""),
             "Build-Time" to buildTime.toString(),
             "Git-Branch" to gitBranch,
@@ -218,6 +_,12 @@
     dependsOn(scanJarForBadCalls)
 }
 
+// Pufferfish start - SIMD Utilities
+tasks.compileJava {
+    options.compilerArgs.add("--add-modules=jdk.incubator.vector")
+}
+// Pufferfish end - SIMD Utilities
+
 // Use TCA for console improvements
 tasks.jar {
     val archiveOperations = services.archiveOperations
@@ -246,7 +_,7 @@
     jvmArgumentProviders.add(provider)
 }
 
-val generatedDir: java.nio.file.Path = layout.projectDirectory.dir("src/generated/java").asFile.toPath()
+val generatedDir: java.nio.file.Path = layout.projectDirectory.dir("../paper-server/src/generated/java").asFile.toPath()
 idea {
     module {
         generatedSourceDirs.add(generatedDir.toFile())
