From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Thu, 20 Mar 2025 22:49:36 -0500
Subject: [PATCH] Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 4e099b584c3435ba1ebce492011d35ae1ac0cdf8..b73e8d6aba63213127dfeece1f3eb331c9782d5d 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -16,6 +16,7 @@ public interface ServerBuildInfo {
      * The brand id for Paper.
      */
     Key BRAND_PAPER_ID = Key.key("papermc", "paper");
+    Key BRAND_PUFFERFISH_ID = Key.key("pufferfish", "pufferfish"); // Pufferfish - Rebrand
 
     /**
      * Gets the {@code ServerBuildInfo}.
diff --git a/src/main/java/org/bukkit/command/defaults/VersionCommand.java b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
index f4dd936d79995014651326b41af706bae179a5a1..d8b18bc3c53d9684ccfea32157eb5c8c1cc34906 100644
--- a/src/main/java/org/bukkit/command/defaults/VersionCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
@@ -257,7 +257,7 @@ public class VersionCommand extends BukkitCommand {
     private void setVersionMessage(final @NotNull Component msg) {
         lastCheck = System.currentTimeMillis();
         final Component message = Component.textOfChildren(
-            Component.text(Bukkit.getVersionMessage(), NamedTextColor.WHITE),
+            gg.pufferfish.pufferfish.PufferfishVersionOverride.getVersionMessage(), // Pufferfish - Rebrand
             Component.newline(),
             msg
         );
