From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul <paul@technove.co>
Date: Tue, 6 Jul 2021 17:51:18 -0500
Subject: [PATCH] Ignore Lookups If Closed


diff --git a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
index 74af9d1b6797e2d697bf1ea67b613ac405a6bc0d..f520eae257dee2e597b416479dcf310e3d90e871 100644
--- a/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
+++ b/src/main/java/org/bukkit/plugin/java/PluginClassLoader.java
@@ -51,6 +51,8 @@ public final class PluginClassLoader extends URLClassLoader implements io.paperm
     private io.papermc.paper.plugin.provider.classloader.PluginClassLoaderGroup classLoaderGroup; // Paper
     public io.papermc.paper.plugin.provider.entrypoint.DependencyContext dependencyContext; // Paper
 
+    private boolean pufferfish$closed = false; // Pufferfish - Ignore lookups if closed
+
     static {
         ClassLoader.registerAsParallelCapable();
     }
@@ -213,7 +215,7 @@ public final class PluginClassLoader extends URLClassLoader implements io.paperm
         }
         Class<?> result = classes.get(name);
 
-        if (result == null) {
+        if (result == null && !this.pufferfish$closed) { // Pufferfish - Ignore lookups if closed
             String path = name.replace('.', '/').concat(".class");
             // Add details to zip file errors - help debug classloading
             JarEntry entry;
@@ -269,6 +271,7 @@ public final class PluginClassLoader extends URLClassLoader implements io.paperm
             this.setClass(name, result); // Paper
         }
 
+        if (result == null) throw new ClassNotFoundException(name); // Pufferfish - Ignore lookups if closed
         return result;
     }
 
@@ -283,6 +286,7 @@ public final class PluginClassLoader extends URLClassLoader implements io.paperm
             // Paper end
             super.close();
         } finally {
+            this.pufferfish$closed = true; // Pufferfish - Ignore lookups if closed
             jar.close();
         }
     }
