From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kevin Raneri <kevin.raneri@gmail.com>
Date: Tue, 8 Feb 2022 12:53:02 -0500
Subject: [PATCH] Pufferfish Optimize Map Rendering


diff --git a/src/main/java/org/bukkit/map/MapPalette.java b/src/main/java/org/bukkit/map/MapPalette.java
index 153414da2bf7704db829cf99f044910ae3643eea..1f131a0586ac6e86159e72102c5a1797db5ad580 100644
--- a/src/main/java/org/bukkit/map/MapPalette.java
+++ b/src/main/java/org/bukkit/map/MapPalette.java
@@ -35,7 +35,7 @@ public final class MapPalette {
     }
 
     @NotNull
-    static final Color[] colors = {
+    public static final Color[] colors = {
         // Start generate - MapPalette#colors
         // @GeneratedFrom 1.21.7-rc1
         new Color(0x00000000, true),
@@ -395,9 +395,15 @@ public final class MapPalette {
         temp.getRGB(0, 0, temp.getWidth(), temp.getHeight(), pixels, 0, temp.getWidth());
 
         byte[] result = new byte[temp.getWidth() * temp.getHeight()];
+        // Pufferfish start - Optimize Map Rendering
+        if (gg.pufferfish.pufferfish.simd.SIMDDetection.isEnabled){
+            gg.pufferfish.pufferfish.simd.VectorMapPalette.matchColorVectorized(pixels, result);
+        } else {
         for (int i = 0; i < pixels.length; i++) {
             result[i] = matchColor(new Color(pixels[i], true));
         }
+        }
+        // Pufferfish end - Optimize Map Rendering
         return result;
     }
 
