From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Fri, 20 Jun 2025 18:07:56 -0500
Subject: [PATCH] Add Sentry


diff --git a/src/main/java/org/bukkit/craftbukkit/scheduler/CraftScheduler.java b/src/main/java/org/bukkit/craftbukkit/scheduler/CraftScheduler.java
index fb05cbd9fba24f1d8ea2bbcc5ef20b2b2964df9f..982455a5eecdb6d80762cc17c2461b467ba5072e 100644
--- a/src/main/java/org/bukkit/craftbukkit/scheduler/CraftScheduler.java
+++ b/src/main/java/org/bukkit/craftbukkit/scheduler/CraftScheduler.java
@@ -478,10 +478,12 @@ public class CraftScheduler implements BukkitScheduler {
                         "Task #%s for %s generated an exception",
                         task.getTaskId(),
                         task.getOwner().getDescription().getFullName());
+                    gg.pufferfish.pufferfish.sentry.SentryContext.setPluginContext(task.getOwner()); // Pufferfish
                     task.getOwner().getLogger().log(
                             Level.WARNING,
                         logMessage,
                             throwable);
+                    gg.pufferfish.pufferfish.sentry.SentryContext.removePluginContext(); // Pufferfish
                     org.bukkit.Bukkit.getServer().getPluginManager().callEvent(
                         new com.destroystokyo.paper.event.server.ServerExceptionEvent(new com.destroystokyo.paper.exception.ServerSchedulerException(logMessage, throwable, task)));
                     // Paper end
