From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Biquaternions <biquaternions@serlith.net>
Date: Thu, 20 Mar 2025 22:46:25 -0500
Subject: [PATCH] Rebrand


diff --git a/net/minecraft/CrashReport.java b/net/minecraft/CrashReport.java
index 84247ddc9e063676c59a6e62da1eb919320bd18f..c4895c7cce724792ce61d84db935aebb74d3b3ed 100644
--- a/net/minecraft/CrashReport.java
+++ b/net/minecraft/CrashReport.java
@@ -30,6 +30,7 @@ public class CrashReport {
     private boolean trackingStackTrace = true;
     private StackTraceElement[] uncategorizedStackTrace = new StackTraceElement[0];
     private final SystemReport systemReport = new SystemReport();
+    private final List<String> extraInfo = List.of("", "DO NOT REPORT THIS TO PAPER! REPORT THIS TO PUFFERFISH!", ""); // Pufferfish - Rebrand
 
     public CrashReport(final String title, final Throwable t) {
         this.title = title;
@@ -129,7 +130,7 @@ public class CrashReport {
     }
 
     public String getFriendlyReport(final ReportType reportType) {
-        return this.getFriendlyReport(reportType, List.of());
+        return this.getFriendlyReport(reportType, this.extraInfo); // Pufferfish - Rebrand
     }
 
     public @Nullable Path getSaveFile() {
@@ -159,7 +160,7 @@ public class CrashReport {
     }
 
     public boolean saveToFile(final Path file, final ReportType reportType) {
-        return this.saveToFile(file, reportType, List.of());
+        return this.saveToFile(file, reportType, this.extraInfo); // Pufferfish - Rebrand
     }
 
     public SystemReport getSystemReport() {
diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 2142757965b7345a011a92de08038473fa8ae320..a54cb80c38e673aacd93b6d4ad2276b854b92706 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1323,7 +1323,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
             // Paper end - Add onboarding message for initial server start
             // Paper start - Improve outdated version checking
             if (System.getProperty("paper.disableStartupVersionCheck") == null && io.papermc.paper.configuration.GlobalConfiguration.get().updateChecker.enabled) {
-                CompletableFuture.runAsync(com.destroystokyo.paper.PaperVersionFetcher::getUpdateStatusStartupMessage, io.papermc.paper.util.MCUtil.ASYNC_EXECUTOR);
+                CompletableFuture.runAsync(gg.pufferfish.pufferfish.PufferfishVersionFetcher::getUpdateStatusStartupMessage, io.papermc.paper.util.MCUtil.ASYNC_EXECUTOR); // Pufferfish - Rebrand
             }
             // Paper end - Improve outdated version checking
 
diff --git a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
index eae4963b6d18c5f92871477b93f81c2c79331cd9..6e06fb1f9907eb4aa48de2247c28dd59672ebd9e 100644
--- a/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
+++ b/net/minecraft/world/level/chunk/storage/RegionFileStorage.java
@@ -309,7 +309,7 @@ public class RegionFileStorage implements AutoCloseable, ca.spottedleaf.moonrise
 
     // Paper start
     private static void printOversizedLog(String msg, Path file, int x, int z) {
-        org.apache.logging.log4j.LogManager.getLogger().fatal(msg + " (" + file.toString().replaceAll(".+[\\\\/]", "") + " - " + x + "," + z + ") Go clean it up to remove this message. /minecraft:tp " + (x<<4)+" 128 "+(z<<4) + " - DO NOT REPORT THIS TO PAPER - You may ask for help on Discord, but do not file an issue. These error messages can not be removed.");
+        org.apache.logging.log4j.LogManager.getLogger().fatal(msg + " (" + file.toString().replaceAll(".+[\\\\/]", "") + " - " + x + "," + z + ") Go clean it up to remove this message. /minecraft:tp " + (x<<4)+" 128 "+(z<<4) + " - DO NOT REPORT THIS TO PUFFERFISH - You may ask for help on Discord, but do not file an issue. These error messages can not be removed."); // Pufferfish - Rebrand
     }
 
     private static CompoundTag readOversizedChunk(RegionFile regionfile, ChunkPos chunkCoordinate) throws IOException {
