From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul <paul@technove.co>
Date: Fri, 2 Jul 2021 18:25:18 -0500
Subject: [PATCH] Reduce Entity Allocations


diff --git a/net/minecraft/world/entity/ai/attributes/AttributeMap.java b/net/minecraft/world/entity/ai/attributes/AttributeMap.java
index 68bad6752bd313ca3ebde7fd4e3b2c46c5dee3ef..f417a0bc6212fc242aa21c0b28dbd39112e45cc8 100644
--- a/net/minecraft/world/entity/ai/attributes/AttributeMap.java
+++ b/net/minecraft/world/entity/ai/attributes/AttributeMap.java
@@ -18,9 +18,11 @@ public class AttributeMap {
     private final Set<AttributeInstance> attributesToSync = new ObjectOpenHashSet<>();
     private final Set<AttributeInstance> attributesToUpdate = new ObjectOpenHashSet<>();
     private final AttributeSupplier supplier;
+    private final java.util.function.Function<Holder<Attribute>, AttributeInstance> pufferfish$createInstance; // Pufferfish
 
     public AttributeMap(final AttributeSupplier supplier) {
         this.supplier = supplier;
+        this.pufferfish$createInstance = key -> this.supplier.createInstance(this::onAttributeModified, key); // Pufferfish
     }
 
     private void onAttributeModified(final AttributeInstance attributeInstance) {
@@ -43,7 +45,7 @@ public class AttributeMap {
     }
 
     public @Nullable AttributeInstance getInstance(final Holder<Attribute> attribute) {
-        return this.attributes.computeIfAbsent(attribute, key -> this.supplier.createInstance(this::onAttributeModified, (Holder<Attribute>)key));
+        return this.attributes.computeIfAbsent(attribute, this.pufferfish$createInstance); // Pufferfish - cache lambda, as for some reason java allocates it anyways
     }
 
     public boolean hasAttribute(final Holder<Attribute> attribute) {
