From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Sauve <paul@technove.co>
Date: Mon, 26 Apr 2021 10:52:56 -0500
Subject: [PATCH] Cache Climbing Check


diff --git a/io/papermc/paper/entity/activation/ActivationRange.java b/io/papermc/paper/entity/activation/ActivationRange.java
index 5cc1e6db2eb3a3a2c3e06ddb97632ff31e3c6833..1d9254715a7005a2afdebc2f7a1884a8d7d32ca6 100644
--- a/io/papermc/paper/entity/activation/ActivationRange.java
+++ b/io/papermc/paper/entity/activation/ActivationRange.java
@@ -239,7 +239,7 @@ public final class ActivationRange {
         }
         // special cases.
         if (entity instanceof final LivingEntity living) {
-            if (living.onClimbable() || living.jumping || living.hurtTime > 0 || !living.activeEffects.isEmpty() || living.isFreezing()) {
+            if (living.pufferfish$onClimbableCached() || living.jumping || living.hurtTime > 0 || !living.activeEffects.isEmpty() || living.isFreezing()) { // Pufferfish - use cached
                 return 1;
             }
             if (entity instanceof final Mob mob && mob.getTarget() != null) {
diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index ff474dec3506a44fc128eff2840ad50b29f5821a..d4b01e37e3ce3721def95d9912f5917e95ebbc2a 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -2174,6 +2174,19 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
         return this.lastClimbablePos;
     }
 
+    // Pufferfish start
+    private boolean pufferfish$cachedOnClimable = false;
+    private BlockPos pufferfish$lastClimbingPosition = null;
+
+    public boolean pufferfish$onClimbableCached() {
+        if (!this.blockPosition().equals(this.pufferfish$lastClimbingPosition)) {
+            this.pufferfish$cachedOnClimable = this.onClimbable();
+            this.pufferfish$lastClimbingPosition = this.blockPosition();
+        }
+        return this.pufferfish$cachedOnClimable;
+    }
+    // Pufferfish end
+
     public boolean onClimbable() {
         if (this.isSpectator()) {
             return false;
