From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Sauve <paul@technove.co>
Date: Mon, 26 Apr 2021 10:52:56 -0500
Subject: [PATCH] Cache Climbing Check


diff --git a/io/papermc/paper/entity/activation/ActivationRange.java b/io/papermc/paper/entity/activation/ActivationRange.java
index bc1317b08c8657f2c045940f2c14e4f251dece0c..4728a672c6caa7727cb992f1976a7a828cbffd7a 100644
--- a/io/papermc/paper/entity/activation/ActivationRange.java
+++ b/io/papermc/paper/entity/activation/ActivationRange.java
@@ -239,7 +239,7 @@ public final class ActivationRange {
         }
         // special cases.
         if (entity instanceof final LivingEntity living) {
-            if (living.onClimbable() || living.jumping || living.hurtTime > 0 || !living.activeEffects.isEmpty() || living.isFreezing()) {
+            if (living.pufferfish$onClimbableCached() || living.jumping || living.hurtTime > 0 || !living.activeEffects.isEmpty() || living.isFreezing()) { // Pufferfish - use cached
                 return 1;
             }
             if (entity instanceof final Mob mob && mob.getTarget() != null) {
diff --git a/net/minecraft/world/entity/LivingEntity.java b/net/minecraft/world/entity/LivingEntity.java
index e7ed26cde4c4e3af8a3dfcf8b221a9f80e1bc4ed..e89cb94d859dde86b3f550a7676e004751532a2b 100644
--- a/net/minecraft/world/entity/LivingEntity.java
+++ b/net/minecraft/world/entity/LivingEntity.java
@@ -2165,6 +2165,19 @@ public abstract class LivingEntity extends Entity implements Attackable, Waypoin
         return this.lastClimbablePos;
     }
 
+    // Pufferfish start
+    private boolean pufferfish$cachedOnClimable = false;
+    private BlockPos pufferfish$lastClimbingPosition = null;
+
+    public boolean pufferfish$onClimbableCached() {
+        if (!this.blockPosition().equals(this.pufferfish$lastClimbingPosition)) {
+            this.pufferfish$cachedOnClimable = this.onClimbable();
+            this.pufferfish$lastClimbingPosition = this.blockPosition();
+        }
+        return this.pufferfish$cachedOnClimable;
+    }
+    // Pufferfish end
+
     public boolean onClimbable() {
         if (this.isSpectator()) {
             return false;
