From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Sauve <paul@technove.co>
Date: Thu, 20 May 2021 12:05:47 -0500
Subject: [PATCH] Better Check For Useless Packets


diff --git a/net/minecraft/server/level/ServerEntity.java b/net/minecraft/server/level/ServerEntity.java
index 3a817c92147bb78c60238f3ddc1644b721aed7a2..3029c369bef0b503dff6b6780bb368fdd2913106 100644
--- a/net/minecraft/server/level/ServerEntity.java
+++ b/net/minecraft/server/level/ServerEntity.java
@@ -181,7 +181,7 @@ public class ServerEntity {
                     packet = ClientboundEntityPositionSyncPacket.of(this.entity);
                     flag3 = true;
                     flag4 = true;
-                } else if ((!flag2 || !flag) && !(this.entity instanceof AbstractArrow)) {
+                } else if (flag2 || flag || this.entity instanceof AbstractArrow) { if ((!flag2 || !flag) && !(this.entity instanceof AbstractArrow)) { // Pufferfish
                     if (flag2) {
                         packet = new ClientboundMoveEntityPacket.Pos(this.entity.getId(), (short)l, (short)l1, (short)l2, this.entity.onGround());
                         flag3 = true;
@@ -194,6 +194,7 @@ public class ServerEntity {
                     flag3 = true;
                     flag4 = true;
                 }
+                } // Pufferfish
 
                 if (this.entity.needsSync || this.trackDelta || this.entity instanceof LivingEntity && ((LivingEntity)this.entity).isFallFlying()) {
                     Vec3 deltaMovement = this.entity.getDeltaMovement();
