From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kevin Raneri <kevin.raneri@gmail.com>
Date: Sat, 11 Dec 2021 22:20:45 -0500
Subject: [PATCH] Throttle Goal Selector On Inactive Ticking


diff --git a/net/minecraft/world/entity/Mob.java b/net/minecraft/world/entity/Mob.java
index 2270ba01201198b031744bbb4a32c1d2cebf254b..8f82b5d31cd98a73e89aaaae39668747b7a1e4a6 100644
--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -215,12 +215,14 @@ public abstract class Mob extends LivingEntity implements Targeting, EquipmentUs
         return this.lookControl;
     }
 
+    int pufferfish$inactiveTickDisableCounter = 0; // Pufferfish - throttle inactive goal selector ticking
     // Paper start
     @Override
     public void inactiveTick() {
         super.inactiveTick();
         if (!this.aware) return; // Paper - Do not tick AI for inactive unaware mobs
-        if (this.goalSelector.pufferfish$inactiveTick(this.pufferfish$activatedPriority, true)) { // Pufferfish - pass activated priority
+        boolean isThrottled = gg.pufferfish.pufferfish.PufferfishConfig.INACTIVE_GOAL_SELECTOR_THROTTLE && pufferfish$inactiveTickDisableCounter++ % 20 != 0; // Pufferfish - throttle inactive goal selector ticking
+        if (this.goalSelector.pufferfish$inactiveTick(this.pufferfish$activatedPriority, true) && !isThrottled) { // Pufferfish - pass activated priroity // Pufferfish - throttle inactive goal selector ticking
             this.goalSelector.tick();
         }
         if (this.targetSelector.pufferfish$inactiveTick(this.pufferfish$activatedPriority, true)) { // Pufferfish - pass activated priority
