From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kevin Raneri <kevin.raneri@gmail.com>
Date: Wed, 27 Dec 2023 22:49:53 -0500
Subject: [PATCH] TPS Catchup


diff --git a/net/minecraft/server/MinecraftServer.java b/net/minecraft/server/MinecraftServer.java
index 095fb68043e50e3f4bc04686e4fa19b6a03983b2..37819607cb4b27d2924eb315bc52346c088c230d 100644
--- a/net/minecraft/server/MinecraftServer.java
+++ b/net/minecraft/server/MinecraftServer.java
@@ -1119,6 +1119,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
         this.scheduledTickStart = this.tickSchedule.getDeadline(interval);
     }
 
+    private long pufferfish$now;
     private void recordEndOfTick() {
         final long prevStart = this.lastTickStart;
         final long currStart = this.currentTickStart;
@@ -1127,6 +1128,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
         this.scheduledTickStart = this.nextTickTimeNanos; // set scheduledStart for next tick
 
         final long now = Util.getNanos();
+        this.pufferfish$now = now;
 
         final ca.spottedleaf.moonrise.common.time.TickTime time = new ca.spottedleaf.moonrise.common.time.TickTime(
             prevStart,
@@ -1335,6 +1337,12 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
                     profilerFiller.popPush("nextTickWait");
                     this.mayHaveDelayedTasks = true;
                     this.delayedTasksMaxNextTickTimeNanos = Math.max(Util.getNanos() + l, this.nextTickTimeNanos);
+                    // Pufferfish start - tps catchup
+                    if (!gg.pufferfish.pufferfish.PufferfishConfig.TPS_CATCHUP) {
+                        this.nextTickTimeNanos = this.pufferfish$now + l;
+                        this.delayedTasksMaxNextTickTimeNanos = nextTickTimeNanos;
+                    }
+                    // Pufferfish end
                     this.startMeasuringTaskExecutionTime();
                     this.recordTaskExecutionTimeWhileWaiting(); // Paper - improve tick loop - record task execution here on MSPT
                     this.finishMeasuringTaskExecutionTime();
